src/GraphQL.Tests/Types/AllScalarGraphTypeTests.cs

```diff
@@ -0,0 +1,807 @@
+            g.CanParseValue("0").ShouldBeFalse();
+            Should.Throw<InvalidOperationException>(() => g.ParseValue("0"));
+            Should.Throw<InvalidOperationException>(() => g.Serialize("0"));
+        }
+
+        [Theory]
+        [InlineData(typeof(ByteGraphType), (byte)0)]
+        [InlineData(typeof(ByteGraphType), (byte)1)]
+        [InlineData(typeof(ByteGraphType), (byte)255)]
+        [InlineData(typeof(SByteGraphType), (sbyte)-128)]
+        [InlineData(typeof(SByteGraphType), (sbyte)0)]
+        [InlineData(typeof(SByteGraphType), (sbyte)127)]
+        [InlineData(typeof(ShortGraphType), short.MinValue)]
+        [InlineData(typeof(ShortGraphType), (short)default)]
+        [InlineData(typeof(ShortGraphType), short.MaxValue)]
+        [InlineData(typeof(UShortGraphType), ushort.MinValue)]
+        [InlineData(typeof(UShortGraphType), ushort.MaxValue)]
+        [InlineData(typeof(IntGraphType), -2000000000)] //float cannot hold the full precision of int
+        [InlineData(typeof(IntGraphType), 0)]
+        [InlineData(typeof(IntGraphType), 2000000000)]
+        [InlineData(typeof(UIntGraphType), 0u)]
+        [InlineData(typeof(UIntGraphType), 4000000000u)]
+        [InlineData(typeof(LongGraphType), -9223300018843156480L)]
+        [InlineData(typeof(LongGraphType), 0L)]
+        [InlineData(typeof(LongGraphType), 9223300018843156480L)]
+        [InlineData(typeof(ULongGraphType), 0ul)]
+        [InlineData(typeof(ULongGraphType), 18446700093244440576uL)]
+        [InlineData(typeof(FloatGraphType), -2.0)]
+        [InlineData(typeof(FloatGraphType), 2.0)]
+        [InlineData(typeof(BigIntGraphType), -1E+25)]
+        [InlineData(typeof(BigIntGraphType), 0)]
+        [InlineData(typeof(BigIntGraphType), 1E+25)]
+        public void parseValue_ok(Type graphType, object value)
+        {
+            if (graphType == typeof(BigIntGraphType))
+                value = new BigInteger(Convert.ToDecimal(value));
+
+            var g = (ScalarGraphType)graphType.GetConstructor(Type.EmptyTypes).Invoke(null);
+            var types = new Type[]
+            {
+                typeof(byte),
+                typeof(sbyte),
+                typeof(short),
+                typeof(ushort),
+                typeof(int),
+                typeof(uint),
+                typeof(long),
+                typeof(ulong),
+                typeof(BigInteger)
+            };
+
+            foreach (var type in types)
+            {
+                object converted;
+                try
+                {
+                    if (type == typeof(BigInteger))
+                    {
+                        converted = new BigInteger((decimal)Convert.ChangeType(value, typeof(decimal)));
+                    }
+                    else
+                    {
+                        converted = Convert.ChangeType(value, type);
+                    }
+                }
+                catch
+                {
+                    continue;
```

Comment:
?

Reply:
e.g. testing the value 500 being passed as a byte is not possible.  so it skips testing the data types where the conversion of the value to the specific data type in the loop fails